batchflow/models/torch/layers/combine.py

```diff
@@ -0,0 +1,173 @@
+
+            self.op = getattr(self, f'{op_name}_forward')
+            self.force_resize = force_resize if force_resize is not None else True
+        elif callable(op):
+            self.op_name = op.__name__
+            self.op = op
+            self.force_resize = force_resize if force_resize is not None else False
+        else:
+            raise ValueError(f'Combine op must be a callable or one from {list(self.OPS.keys())}, got {op} instead!')
+
+    def forward(self, inputs):
+        # Inputs
+        self.input_ids = [id(item) for item in inputs]
+        if self.idx != 0:
+            inputs[0], inputs[self.idx] = inputs[self.idx], inputs[0]
+        self.after_ids = [id(item) for item in inputs]
+        self.input_shapes = get_shape(inputs)
+
+        # Resize
+        if self.force_resize:
+            inputs = self.spatial_resize(inputs)
+            self.resized_shapes = get_shape(inputs)
+
+        # Outputs
+        output = self.op(inputs)
+        self.output_shapes = get_shape(output)
+        return output
+
+    def extra_repr(self):
+        """ Report shapes before and after combination to a repr. """
+        res = f'op={"callable " if not isinstance(self.name, str) else ""}{self.op_name}'
+        res += f', leading_idx={self.idx}, force_resize={self.force_resize}'
+        for key, value in self.kwargs.items():
+            res += f', {key}={value}'
+
+        if getattr(self, 'verbosity', 10) > 2:
+            res += f',\ninput_shapes={self.input_shapes}'
+
+            if self.force_resize:
+                res += f',\nresized_shapes={self.resized_shapes}'
+
+            res += f',\noutput_shapes={self.output_shapes}'
+
+            res += f',\ninput_ids={self.input_ids}'
+            res += f',\nafter_ids={self.after_ids}'
+        return res
+
+
+    def spatial_resize(self, inputs):
+        """ Force the same shapes of the inputs, if needed. """
+        shape_ = get_shape(inputs[0])
+        dim_ = get_num_dims(inputs[0])
+        spatial_shape_ = shape_[-dim_:]
+
+        resized = []
+        for item in inputs:
+            shape = get_shape(item)
+            dim = get_num_dims(item)
+            spatial_shape = shape[-dim:]
+            if dim > 0 and spatial_shape != tuple([1]*dim) and spatial_shape != spatial_shape_:
+                item = Crop(inputs[0])(item)
+            resized.append(item)
+        return resized
+
+    def concat_forward(self, inputs):
+        return torch.cat(inputs, dim=1)
+
+    def sum_forward(self, inputs):
+        """ Addition with broadcasting. """
+        result = 0
+        for item in inputs:
+            result = result + item
+        return result
+
+    def mul_forward(self, inputs):
+        """ Multiplication with broadcasting. """
+        result = 1
+        for item in inputs:
+            result = result * item
+        return result
+
+    def mean_forward(self, inputs):
+        return torch.mean(inputs)
```

Comment:
What is the `inputs` dtype - tensor or list? `torch.mean` input must be a tensor.